src/main/java/HelloJavaMesh.java

```diff
@@ -79,20 +81,18 @@ public HelloJavaMesh() throws Exception {
     private void handlePeerChanged(MeshManager.RightMeshEvent e) {
         // Update peer list.
         MeshManager.PeerChangedEvent event = (MeshManager.PeerChangedEvent) e;
+
         if (event.state != REMOVED && !users.contains(event.peerUuid)) {
             users.add(event.peerUuid);
             System.out.println("PEER JOINED: " + event.peerUuid);
-        } else if (event.state == REMOVED){
+        } else if (event.state == REMOVED && users.contains(event.peerUuid)) {
```

Comment:
doesn't the remove check for this? or will it return exception if it's not in the set?